### ISSUE TITLE
Need for cropImage(image,path) canvas flutter.

### ISSUE THREAD

**Issue by abhishekove** [flutter/flutter#68984]
There is a need for a function like crop images that crop the image and resizes it according to actual image width and height. the function similar to the clip image.

**Comment by pedromassangocode** [comment]
Hi @abhishekove 
You need to be clear while posting a proposal for Flutter.

Where this cropImage should be added?
How it should exactly work?
Should it return the cropped image?

Please elaborate more.

**Comment by abhishekove** [comment]
According to me cropImage must be in canvas and must return cropped and transformed image.

**Comment by pedromassangocode** [comment]
I'm still not sure what you are asking exactly so will leave this opened for the right person to check.

Cc @dnfield

**Comment by dnfield** [comment]
It would be helpful to know what exactly you're trying to do, and what you've tried so far, and how it's failing.

You may be interested in:

https://api.flutter.dev/flutter/dart-ui/Canvas/drawImageRect.html

Which you could use to draw a cropped portion of an image and then use something like https://api.flutter.dev/flutter/dart-ui/Picture/toImage.html to rasterize it again.

There are no plans to add any additional API to Flutter for scaling/cropping images - but there should be enough base API to write a package to do this if you want.

**Comment by pedromassangocode** [comment]
@abhishekove Does the information in https://github.com/flutter/flutter/issues/68984#issuecomment-717065470 helps you solve your use-case?

**Comment by abhishekove** [comment]
I want something like this.
[https://github.com/flutter-clutter/flutter-simple-edge-detection](url)

**Comment by ydotmalik** [comment]
@dnfield drawImageRect does not crop an image. For example, in the following:
```
canvas.drawImageRect(
 image,
 Rect.fromLTWH(0, 0, 2400, 3000),
 Rect.fromLTWH(500, 500, 1000, 1000), Paint())
```

Just draws the entire 2400x3000 image at offset (500, 500) but scaled to fit a 1000x1000 area.

**Comment by dnfield** [comment]
You are right. I think I meant to link drawAtlas, which should work.  Otherwise using a cliprect with drawImage should work
